local/compose/attach.go

```diff
@@ -42,6 +42,7 @@ func (s *composeService) attach(ctx context.Context, project *types.Project, con
 	if err != nil {
 		return nil, err
 	}
+	containers = Containers(containers).filter(isService(project.ServiceNames()...))
```

Comment:
Nit: could we add this filtering next to `projectFilter(project.Name)` for more clarity ? I guess it would require a nice way to express and OR condition, that we don't have ?

Reply:
`projectFilter` produces a moby API filter, unrelated to `Containers.filter` which use forceps to tweak golang into something functional-ish  :P